test/Microsoft.Health.Dicom.Web.Tests.E2E/Functions/HostJsonFileConfigurationSource.cs

```diff
@@ -0,0 +1,198 @@
+                    case JTokenType.Raw:
+                    case JTokenType.Bytes:
+                    case JTokenType.TimeSpan:
+                        string key = AzureFunctionsConfiguration.JobHostSectionName + ConfigurationPath.KeyDelimiter + ConfigurationPath.Combine(_path.Reverse());
+                        Data[key] = token.Value<JValue>().ToString(CultureInfo.InvariantCulture);
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            private void ProcessArray(JArray jArray)
+            {
+                for (int i = 0; i < jArray.Count; i++)
+                {
+                    _path.Push(i.ToString(CultureInfo.InvariantCulture));
+                    ProcessToken(jArray[i]);
+                    _path.Pop();
+                }
+            }
+
+            /// <summary>
+            /// Read and apply host.json configuration.
+            /// </summary>
+            private JObject LoadHostConfigurationFile()
+            {
+                // Before configuration has been fully read, configure a default logger factory
+                // to ensure we can log any configuration errors. There's no filters at this point,
+                // but that's okay since we can't build filters until we apply configuration below.
+                // We'll recreate the loggers after config is read. We initialize the public logger
+                // to the startup logger until we've read configuration settings and can create the real logger.
+                // The "startup" logger is used in this class for startup related logs. The public logger is used
+                // for all other logging after startup.
+
+                string hostFilePath = Path.Combine(_configurationSource.ScriptPath, ScriptConstants.HostMetadataFileName);
+                JObject hostConfigObject = LoadHostConfig(hostFilePath);
+                hostConfigObject = InitializeHostConfig(hostFilePath, hostConfigObject);
+
+                return hostConfigObject;
+            }
+
+            private static JObject InitializeHostConfig(string hostJsonPath, JObject hostConfigObject)
+            {
+                // If the object is empty, initialize it to include the version and write the file.
+                if (!hostConfigObject.HasValues)
+                {
+                    hostConfigObject = GetDefaultHostConfigObject();
+                    TryWriteHostJson(hostJsonPath, hostConfigObject);
+                }
+
+                string hostJsonVersion = hostConfigObject["version"]?.Value<string>();
+                if (string.IsNullOrEmpty(hostJsonVersion))
+                {
+                    throw new FormatException($"The {ScriptConstants.HostMetadataFileName} file is missing the required 'version' property. See https://aka.ms/functions-hostjson for steps to migrate the configuration file.");
+                }
+
+                if (!hostJsonVersion.Equals("2.0"))
```

Comment:
Should this be part of ScriptConstants?

Reply:
Sure! A lot of this was just ripped from the Azure Functions host source code and reduced to what we need to run locally lol